### PR TITLE
Feat/incubator dialog fixes and improvements

### PR DESCRIPTION
**Pull request by M-i-k-e-l**
## Description
You can look at each commit separately (except the last 3 commits).
Created ImperativeDialog to simplify the code and fix a bug (click on ListItem).
Add alignment support (bottom, center etc.).
Support `useSafeArea`.

Related ticket: WOAUILIB-2317

## Changelog
Incubator.Dialog - fix closing imperatively (button click), add alignment support (bottom, center etc.), support `useSafeArea`

### CODE REVIEW

File: src/incubator/Dialog/helpers/useSafeAreaView.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  const topSafeArea = useMemo(() => {
+    const hasTopSafeArea = useSafeArea && alignmentType !== AlignmentType.BOTTOM;
+    if (hasTopSafeArea) {
+      return <SafeAreaSpacerView/>;
+    }
+  }, [useSafeArea, alignmentType]);
+
+  const bottomSafeArea = useMemo(() => {
+    const hasBottomSafeArea = Constants.isIphoneX && useSafeArea && alignmentType === AlignmentType.BOTTOM;
```
Comment (lidord-wix) [inline src/incubator/Dialog/helpers/useSafeAreaView.tsx:28]: is `Constants.isIphoneX` covers all the devices with a safe area?

**Review by lidord-wix — COMMENTED** [review]
Looks good overall,
I left few small comments..

File: demo/src/screens/incubatorScreens/IncubatorDialogScreen.tsx [hunk 2]
```diff
@@ -65,7 +70,14 @@ export default class IncubatorDialogScreen extends Component {
...
           <Button marginV-20 label="Open Dialog" onPress={this.openDialog}/>
         </View>
-        <Incubator.Dialog visible={visible} onDismiss={this.closeDialog} bottom containerStyle={styles.dialogContainer}>
+        <Incubator.Dialog
+          useSafeArea
+          visible={visible}
+          onDismiss={this.onDismiss}
+          bottom
+          centerH
+          modalProps={this.modalProps}
```
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply demo/src/screens/incubatorScreens/IncubatorDialogScreen.tsx:79]: No, you can try to: `command + shift + p` --> `restart TS server`.
BTW, I think that testing by running `npm run build:dev` is the best way to know for sure (although sometimes a `clean-install` is required :/ ).

File: src/incubator/Dialog/types.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import {PropsWithChildren} from 'react';
+import {AlignmentModifiers} from '../../commons/modifiers';
+import {ModalProps} from '../../components/modal';
+import {ViewProps} from '../../components/view';
+import {PanningDirections, PanningDirectionsEnum} from '../panView';
+type DialogDirections = PanningDirections;
+const DialogDirectionsEnum = PanningDirectionsEnum;
+export {DialogDirections, DialogDirectionsEnum};
+
+export interface _DialogProps extends AlignmentModifiers, Pick<ViewProps, 'useSafeArea'> {
```
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/incubator/Dialog/types.ts:10]: I get an TS error in `ImperativeDialog` if I remove the export.

**Review by lidord-wix — APPROVED** [review]

**Pull request merged**